packages/coreutils/src/statedb.ts

```diff
@@ -203,118 +155,95 @@ export class StateDB<T extends ReadonlyJSONValue = ReadonlyJSONValue>
    * requirement for `fetch()`, `remove()`, and `save()`, it *is* necessary for
    * using the `list(namespace: string)` method.
    */
-  save(id: string, value: T): Promise<void> {
-    return this._ready.then(() => {
-      this._save(id, value);
-      this._changed.emit({ id, type: 'save' });
-    });
+  async save(id: string, value: T): Promise<void> {
+    await this._ready;
+    await this._save(id, value);
+    this._changed.emit({ id, type: 'save' });
   }
 
   /**
    * Return a serialized copy of the state database's entire contents.
    *
    * @returns A promise that bears the database contents as JSON.
    */
-  toJSON(): Promise<{ [id: string]: T }> {
-    return this._ready.then(() => {
-      const prefix = `${this._window}:${this.namespace}:`;
-      const mask = (key: string) => key.replace(prefix, '');
+  async toJSON(): Promise<{ readonly [id: string]: T }> {
+    await this._ready;
 
-      return Private.toJSON<T>(prefix, mask);
-    });
+    const { ids, values } = await this._list();
+
+    return values.reduce(
+      (acc, val, idx) => {
```

Comment:
One day I will be comfortable reading `reduce` functions. But it is not this day.